python/prophet/forecaster.py

```diff
@@ -1563,6 +1748,12 @@ def make_future_dataframe(self, periods, freq='D', include_history=True):
         """
         if self.history_dates is None:
             raise Exception('Model has not been fit.')
+        if freq is None:
+            # taking the tail makes freq inference more reliable
+            freq = pd.infer_freq(self.history_dates.tail(5))
+            # returns None if inference failed
+            if freq is None:
+                raise Exception('Unable to infer `freq`')
```

Comment:
I think we'd rather encourage the user to provide a `freq`, or just default to `'D'`, so I'd remove this code.

Reply:
One of my biggest nuisances with Prophet has always been the need to make_future_dataframe when it isn't that hard to just automatically detect frequency and make it. Sure 1% of the time it will be wrong but mostly it is helpful.

That said, I don't really care, so happy to remove this entire addition